src/services/web3/web3.service.ts

```diff
@@ -1,20 +1,42 @@
-import { JsonRpcProvider } from '@ethersproject/providers';
+import { ComputedRef, computed } from 'vue';
+import { Contract } from '@ethersproject/contracts';
+import { ErrorCode } from '@ethersproject/logger';
+import {
+  JsonRpcProvider,
+  TransactionResponse,
+  Web3Provider
+} from '@ethersproject/providers';
 import { rpcProviderService as _rpcProviderService } from '../rpc-provider/rpc-provider.service';
+import { logFailedTx } from '@/lib/utils/logging';
+import { gasPriceService } from '@/services/gas-price/gas-price.service';
+import ConfigService, { configService } from '@/services/config/config.service';
+import { MetamaskError } from '@/types';
 
 interface Web3Profile {
   ens: string | null;
 }
 
+const RPC_INVALID_PARAMS_ERROR_CODE = -32602;
+const EIP1559_UNSUPPORTED_REGEX = /network does not support EIP-1559/i;
+
 export default class Web3Service {
-  provider: JsonRpcProvider;
+  appProvider: ComputedRef<JsonRpcProvider>;
+  userProvider!: ComputedRef<Web3Provider>;
+
+  constructor(
+    private readonly rpcProviderService = _rpcProviderService,
+    private readonly config: ConfigService = configService
+  ) {
+    this.appProvider = computed(() => this.rpcProviderService.jsonProvider);
```

Comment:
For now at least, the appProvider/jsonProvider is static and since `this.rpcProviderService.jsonProvider` is not a reactive property there is no point setting appProvider as a computed property because it will never update.

Also, although I think it's ok to pass in reactive properties from the frontend as inputs, I think we should avoid defining reactive properties within services.

Reply:
Makes sense